src/main/java/com/microsoft/azure/vmagent/util/LocationCache.java

```diff
@@ -0,0 +1,59 @@
+package com.microsoft.azure.vmagent.util;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.resources.Provider;
+import com.microsoft.azure.management.resources.ProviderResourceType;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public final class LocationCache {
+    private static Map<String, Set<String>> regions = new HashMap<>();
+
+    public static Set<String> getLocation(Azure azureClient, String serviceManagementURL) throws Exception {
+        if (regions.containsKey(serviceManagementURL) && !regions.get(serviceManagementURL).isEmpty()) {
```

Comment:
Is `static` means that it won't be invalidated until a Jenkins restart? Will it be a problem if user's Jenkins won't restart for a long time?

Reply:
Shall we add an expire time?